### ISSUE TITLE
Using `Expr::field` panics 

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#10565]
### Describe the bug

After https://github.com/apache/datafusion/pull/10375 merged [`Expr::field`](https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.Expr.html#method.field) now panics if you use it (as we did in influxdb) DataFusion panics when you try to execute it

### To Reproduce

Try to evaluate an expression like `col("props").field("a")`

Here is a full reproducer in the `sql_integration` test:

```diff
(venv) andrewlamb@Andrews-MacBook-Pro:~/Software/datafusion$ git diff
diff --git a/datafusion/core/tests/expr_api/mod.rs b/datafusion/core/tests/expr_api/mod.rs
index d7e839824..d4141a836 100644
--- a/datafusion/core/tests/expr_api/mod.rs
+++ b/datafusion/core/tests/expr_api/mod.rs
@@ -58,6 +58,25 @@ fn test_eq_with_coercion() {
     );
 }

+
+#[test]
+fn test_expr_field() {
+    // currently panics with
+    // Internal("NamedStructField should be rewritten in OperatorToFunction")
+    evaluate_expr_test(
+        col("props").field("a"),
+        vec![
+            "+------------+",
+            "| expr       |",
+            "+------------+",
+            "| 2021-02-01 |",
+            "| 2021-02-02 |",
+            "| 2021-02-03 |",
+            "+------------+",
+        ],
+    );
+}
+
```

### Expected behavior

Ideally the test should pass `Expr::field` would continue to work 

We could also potentially remove `Expr::field` but I think that would be less user friendly

### Additional context

I am pretty sure I `Expr::field` is widely used so I think we should continue to support it if possible

**Comment by alamb** [comment]
I wonder if we could have Expr::field call get_field if the core functions feature was enabled and panic otherwise 🤔

That would be easy to use for most people and backwards compatible

**Comment by jayzhan211** [comment]
I can fix this along with #10374

**Comment by alamb** [comment]
Thank you @jayzhan211 🙏  -- I will review it now
